### ISSUE TITLE
PyPDF2.utils.PdfReadError: Unexpected escaped string: G

### ISSUE THREAD

**Issue by retsyo** [py-pdf/PyPDF2#444]
I have tested Python 2.7 and Python 3.6

I just use the code from https://stackoverflow.com/a/465901 (or you can read it at the very end of this issue) to crop PDF files. Some PDF works fine, but one PDF (sorry, I cannot give this PDF file) says
```shell
(531, 666)
......
(531.36, 666)
Traceback (most recent call last):
  File "crop_pdf.py", line 23, in <module>
    output.write(out_f)
  File "R:\PyPDF2\pdf.py", line 482, in write
    self._sweepIndirectReferences(externalReferenceMap, self._root)
  File "R:\PyPDF2\pdf.py", line 572, in _sweepIndirectReferences
    self._sweepIndirectReferences(externMap, realdata)
  File "R:\PyPDF2\pdf.py", line 548, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "R:\PyPDF2\pdf.py", line 572, in _sweepIndirectReferences
    self._sweepIndirectReferences(externMap, realdata)
  File "R:\PyPDF2\pdf.py", line 548, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "R:\PyPDF2\pdf.py", line 557, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, data[i])
  File "R:\PyPDF2\pdf.py", line 572, in _sweepIndirectReferences
    self._sweepIndirectReferences(externMap, realdata)
  File "R:\PyPDF2\pdf.py", line 548, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "R:\PyPDF2\pdf.py", line 548, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "R:\PyPDF2\pdf.py", line 548, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "R:\PyPDF2\pdf.py", line 580, in _sweepIndirectReferences
    newobj = data.pdf.getObject(data)
  File "R:\PyPDF2\pdf.py", line 1676, in getObject
    retval = readObject(self.stream, self)
  File "R:\PyPDF2\generic.py", line 66, in readObject
    return DictionaryObject.readFromStream(stream, pdf)
  File "R:\PyPDF2\generic.py", line 581, in readFromStream
    value = readObject(stream, pdf)
  File "R:\PyPDF2\generic.py", line 77, in readObject
    return readStringFromStream(stream)
  File "R:\PyPDF2\generic.py", line 392, in readStringFromStream
    raise utils.PdfReadError(r"Unexpected escaped string: %s" % tok)
PyPDF2.utils.PdfReadError: Unexpected escaped string: G
```

```python
from PyPDF2 import PdfFileWriter, PdfFileReader

with open("Nim in Action.pdf", "rb") as in_f:
    input1 = PdfFileReader(in_f)
    output = PdfFileWriter()

    numPages = input1.getNumPages()
    print( "document has %s pages." % numPages)

    for i in range(numPages):
        page = input1.getPage(i)
        print(page.mediaBox.getUpperRight_x(), page.mediaBox.getUpperRight_y())
        page.trimBox.lowerLeft = (25, 25)
        page.trimBox.upperRight = (225, 225)
        page.cropBox.lowerLeft = (50, 50)
        page.cropBox.upperRight = (200, 200)
        output.addPage(page)

    with open("Nim in Action-out.pdf", "wb") as out_f:
        output.write(out_f)
```

**Comment by btbouwens** [comment]
I have a similar situation here. Installed:
  krop-0.5.1-10.fc34.noarch              python3-PyPDF2-1.26.0-16.fc34.noarch              python3-krop-0.5.1-10.fc34.noarch              python3-poppler-qt5-0.75.0-7.fc34.x86_64   
  
then when I try to write a cropped PDF:
```Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/krop/mainwindow.py", line 340, in slotKrop
    raise err
  File "/usr/lib/python3.9/site-packages/krop/mainwindow.py", line 319, in slotKrop
    cropper.writeToStream(fp.file)
  File "/usr/lib/python3.9/site-packages/krop/pdfcropper.py", line 81, in writeToStream
    self.output.write(stream)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 482, in write
    self._sweepIndirectReferences(externalReferenceMap, self._root)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 571, in _sweepIndirectReferences
    self._sweepIndirectReferences(externMap, realdata)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 547, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 571, in _sweepIndirectReferences
    self._sweepIndirectReferences(externMap, realdata)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 547, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 556, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, data[i])
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 571, in _sweepIndirectReferences
    self._sweepIndirectReferences(externMap, realdata)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 547, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, value)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 556, in _sweepIndirectReferences
    value = self._sweepIndirectReferences(externMap, data[i])
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 577, in _sweepIndirectReferences
    newobj = data.pdf.getObject(data)
  File "/usr/lib/python3.9/site-packages/PyPDF2/pdf.py", line 1611, in getObject
    retval = readObject(self.stream, self)
  File "/usr/lib/python3.9/site-packages/PyPDF2/generic.py", line 66, in readObject
    return DictionaryObject.readFromStream(stream, pdf)
  File "/usr/lib/python3.9/site-packages/PyPDF2/generic.py", line 579, in readFromStream
    value = readObject(stream, pdf)
  File "/usr/lib/python3.9/site-packages/PyPDF2/generic.py", line 66, in readObject
    return DictionaryObject.readFromStream(stream, pdf)
  File "/usr/lib/python3.9/site-packages/PyPDF2/generic.py", line 579, in readFromStream
    value = readObject(stream, pdf)
  File "/usr/lib/python3.9/site-packages/PyPDF2/generic.py", line 77, in readObject
    return readStringFromStream(stream)
  File "/usr/lib/python3.9/site-packages/PyPDF2/generic.py", line 390, in readStringFromStream
    raise utils.PdfReadError(r"Unexpected escaped string: %s" % tok)
PyPDF2.utils.PdfReadError: Unexpected escaped string: b'm'
```

**Comment by btbouwens** [comment]
I'm trying to find back the file. In the mean time I found other files producing other PdfReadError's but those are outside the scope of this issue I think:

-   File "/usr/lib/python3.10/site-packages/PyPDF2/pdf.py", line 1631, in getObject
    raise utils.PdfReadError("Could not find object.")
-   File "/usr/lib/python3.10/site-packages/PyPDF2/pdf.py", line 1617, in getObject
    raise utils.PdfReadError("file has not been decrypted")
-   File "/usr/lib/python3.10/site-packages/PyPDF2/pdf.py", line 1901, in read
    raise utils.PdfReadError("Could not find xref table at specified location")

**Comment by btbouwens** [comment]
I found back the file, it's a little schematic:
[TPX3_Bias.pdf](https://github.com/py-pdf/PyPDF2/files/9018339/TPX3_Bias.pdf)
The issue appears when I run `krop -o ~/kroptest/TPX3_Bias.pdf --autotrim --go TPX3_Bias.pdf`
Some version info:
* krop-0.5.1-12.fc35.noarch
* python3-krop-0.5.1-12.fc35.noarch
* python3-3.10.5-2.fc35.x86_64
* python3-PyPDF2-1.26.0-18.fc35.noarch
* poppler-21.08.0-1.fc35.x86_64
* poppler-qt5-21.08.0-1.fc35.x86_64
* python3-poppler-qt5-21.1.0-4.fc35.x86_64

Hope this helps

**Comment by MartinThoma** [comment]
Interesting, so I guess you're using https://github.com/arminstraub/krop/tree/master/krop ?

Could you please share your PyPDF2 version? You can get it via:

```
python -c "import PyPDF2;print(PyPDF2.__version__)" 
``

**Comment by btbouwens** [comment]
Indeed, but actually a somewhat outdated version (0.5.1) of krop that came with Fedora; maybe I should try 0.6.0.

`python -c "import PyPDF2;print(PyPDF2.__version__)"` gives 1.26, as expected using python3-PyPDF2-1.26.0-18.fc35.noarch

**Comment by btbouwens** [comment]
So I noticed that my PyPDF2 is kind of old, and tried:
```
% pip3 install PyPDF2 -I --user
% python3 -c "import PyPDF2;print(PyPDF2.__version__)"                         

2.4.1
% pip3 install https://github.com/arminstraub/krop/archive/v0.6.0.tar.gz --user
% ~/.local/bin/krop -o bladiebla.pdf --trim --go TPX3_Bias.pdf     
Unexpected escaped string: m
Unexpected escaped string: d
Unexpected escaped string: i
Unexpected escaped string: N
GPL Ghostscript 9.55.0 (2021-09-27)
Copyright (C) 2021 Artifex Software, Inc.  All rights reserved.
This software is supplied under the GNU AGPLv3 and comes with NO WARRANTY:
see the file COPYING for details.
Processing pages 1 through 1.
Page 1
Loading NimbusSans-Regular font from /usr/share/ghostscript/Resource/Font/NimbusSans-Regular... 5086980 3558514 2979232 1556919 4 done.
Loading NimbusMonoPS-Bold font from /usr/share/ghostscript/Resource/Font/NimbusMonoPS-Bold... 5153092 3780727 3120632 1695229 4 done.
```
And despite the messages, I do get a readable output file.

**Comment by MartinThoma** [comment]
Nice!

That message is just a warning, informing the user that something about the PDF is broken (end thus might result in different than expected output).

**Comment by MartinThoma** [comment]
As this seems not to be an issue anymore, I'll close the ticket. Please let me know if that changes / if I missunderstood :-)

**Comment by btbouwens** [comment]
It actually leaves one thing: persuading Fedora to update the version shipped with it: even Fedora 37 has PyPDF 1.26.

**Comment by btbouwens** [comment]
No, I don't know. I found https://docs.fedoraproject.org/en-US/package-maintainers/Package_Update_Guide/ but I'm not sure if/how that applies.
